tournamentcontrol/competition/tests/test_logo_fields.py

```diff
@@ -0,0 +1,215 @@
+        img = Image.new("RGB", (100, 100), color="red")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="PNG")
+        img_bytes.seek(0)
+        cls.png_data = img_bytes.read()
+        
+        # Create a simple JPEG image
+        img = Image.new("RGB", (100, 100), color="blue")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="JPEG")
+        img_bytes.seek(0)
+        cls.jpg_data = img_bytes.read()
+        
+        # Create a simple SVG (text-based)
+        cls.svg_data = b'<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"></svg>'
+        
+        # Create invalid file (PDF)
+        cls.pdf_data = b"%PDF-1.4 fake pdf content"
+
+    def test_competition_has_logo_fields(self):
+        """Verify Competition model has logo_colour and logo_monochrome fields."""
+        competition = CompetitionFactory()
+        self.assertTrue(hasattr(competition, "logo_colour"))
+        self.assertTrue(hasattr(competition, "logo_monochrome"))
+
+    def test_club_has_logo_fields(self):
+        """Verify Club model has logo_colour and logo_monochrome fields."""
+        club = ClubFactory()
+        self.assertTrue(hasattr(club, "logo_colour"))
+        self.assertTrue(hasattr(club, "logo_monochrome"))
+
+    def test_season_has_logo_fields(self):
+        """Verify Season model has logo_colour and logo_monochrome fields."""
+        season = SeasonFactory()
+        self.assertTrue(hasattr(season, "logo_colour"))
+        self.assertTrue(hasattr(season, "logo_monochrome"))
+
+    def test_division_has_logo_fields(self):
+        """Verify Division model has logo_colour and logo_monochrome fields."""
+        division = DivisionFactory()
+        self.assertTrue(hasattr(division, "logo_colour"))
+        self.assertTrue(hasattr(division, "logo_monochrome"))
+
+    def test_team_has_logo_fields(self):
+        """Verify Team model has logo_colour and logo_monochrome fields."""
+        team = TeamFactory()
+        self.assertTrue(hasattr(team, "logo_colour"))
+        self.assertTrue(hasattr(team, "logo_monochrome"))
+
+    def test_logo_fields_optional(self):
+        """Verify logo fields are optional (blank=True, null=True)."""
+        competition = CompetitionFactory()
+        # ImageField.name is None when no file is uploaded
+        self.assertIsNone(competition.logo_colour.name)
+        self.assertIsNone(competition.logo_monochrome.name)
+
+    def test_competition_logo_accepts_png(self):
+        """Test that Competition accepts PNG files for logos."""
+        competition = CompetitionFactory()
+        png_file = SimpleUploadedFile(
+            "competition_logo.png", self.png_data, content_type="image/png"
+        )
+        
+        # Assign logo and validate
+        competition.logo_colour = png_file
+        competition.full_clean()
+        competition.save()
+        competition.refresh_from_db()
+        
+        self.assertTrue(competition.logo_colour.name)
```

Comment:
Another point missed.

Test this through the admin view. It's clearer and simpler, and is the way users will use the code, so needs doing anyway!